zebra-network/src/peer/connection/tests.rs

```diff
@@ -1,3 +1,64 @@
 //! Tests for peer connections
 
+use futures::channel::mpsc;
+use tokio::io::{duplex, DuplexStream};
+use tokio_util::codec::{FramedRead, FramedWrite};
+
+use zebra_chain::parameters::Network;
+use zebra_test::mock_service::MockService;
+
+use crate::{
+    peer::{
+        client::ClientRequestReceiver, connection::State, ClientRequest, Connection, ErrorSlot,
+    },
+    peer_set::ActiveConnectionCounter,
+    protocol::external::Codec,
+    Request, Response,
+};
+
+mod prop;
 mod vectors;
+
+/// Creates a new [`Connection`] instance for testing.
+fn new_test_connection<A>() -> (
+    Connection<MockService<Request, Response, A>, FramedWrite<DuplexStream, Codec>>,
+    mpsc::Sender<ClientRequest>,
+    MockService<Request, Response, A>,
```

Comment:
The `MockService` has a generic `A` assertion type parameter. This is different for property tests and unit tests.

The `vector` and `prop` modules have `new_test_connection` functions that specify which assertion type should be used.